wgpu-types/src/lib.rs

```diff
@@ -2097,6 +2097,16 @@ impl TextureFormatFeatureFlags {
             _ => false,
         }
     }
+
+    /// A `Vec` of supported sample counts.
+    pub fn supported_sample_counts(&self) -> Vec<u32> {
+        let all_possible_sample_counts: Vec<u32> = vec![1, 2, 4, 8, 16];
+        all_possible_sample_counts
+            .iter()
+            .filter(|sc| self.sample_count_supported(**sc))
+            .cloned()
+            .collect()
+    }
```

Comment:
```suggestion
    pub fn supported_sample_counts(&self) -> Vec<u32> {
        let all_possible_sample_counts: [u32; 5] = [1, 2, 4, 8, 16];
        all_possible_sample_counts
            .into_iter()
            .filter(|&sc| self.sample_count_supported(sc))
            .collect()
    }
```

Small nit here, will auto-apply